src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -432,6 +442,31 @@ object PruneDeadFields {
           rowType = unify(child.typ.rowType, rowDep),
           globalType = requestedType.globalType)
         memoizeTableIR(child, requestedChildType, memo)
+      case MatrixAnnotateRowsTable(child, table, uid, key) =>
+        val fieldDep = requestedType.rvRowType.fieldOption(uid).map(_.typ.asInstanceOf[TStruct])
+        fieldDep match {
+          case Some(struct) =>
+            val tableDep = table.typ.copy(rowType = unify(
+              table.typ.rowType,
+              FastIndexedSeq[TStruct](table.typ.rowType.filterSet(table.typ.key.get.toSet, true)._1) ++
+              FastIndexedSeq(struct): _*),
+              globalType = minimal(table.typ.globalType))
+            memoizeTableIR(table,tableDep, memo)
+            val keyDep = unify(child.typ,
+              key.map(_.map(ir => memoizeAndGetDep(ir, ir.typ, child.typ, memo)))
+                .getOrElse(FastIndexedSeq.empty[MatrixType]): _*)
+            val matDep = unify(
+              child.typ,
+              keyDep,
+              requestedType.copy(rvRowType =
+                unify(child.typ.rvRowType,
+                  minimal(child.typ).rvRowType,
```

Comment:
is the minimal here to prevent the child from being completely removed if no fields survive into the requested type?

Reply:
this is to make sure we don't blow away key fields and get assertion errors on the MatrixType constructor